cmd/snap-confine/mount-support.c

```diff
@@ -330,16 +333,49 @@ static void sc_bootstrap_mount_namespace(const struct sc_mount_config *config)
 		};
 		for (const char **dirs = dirs_from_core; *dirs != NULL; dirs++) {
 			const char *dir = *dirs;
+			struct stat buf;
 			if (access(dir, F_OK) == 0) {
 				sc_must_snprintf(src, sizeof src, "%s%s",
 						 config->rootfs_dir, dir);
 				sc_must_snprintf(dst, sizeof dst, "%s%s",
 						 scratch_dir, dir);
-				sc_do_mount(src, dst, NULL, MS_BIND, NULL);
-				sc_do_mount("none", dst, NULL, MS_SLAVE, NULL);
+				if (lstat(src, &buf) == 0
+				    && lstat(dst, &buf) == 0) {
+					sc_do_mount(src, dst, NULL, MS_BIND,
+						    NULL);
+					sc_do_mount("none", dst, NULL, MS_SLAVE,
+						    NULL);
+				}
 			}
 		}
 	}
+	if (config->uses_base_snap) {
+		// snapd dir: we need it so that we can actually run
+		// anything because snap-exec needs to be visible the
+		// base snap
```

Comment:
Seems slightly misleading. We don't want snap-exec to ship in base snaps, right?

Reply:
Indeed, I reworded the comment now to make more clear what is needed.